sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosExcludedRegions.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+/**
+ * A class which encapsulates a set of excluded regions.
+ * */
+public final class CosmosExcludedRegions {
+    private final Set<String> excludedRegions;
+    private final String excludedRegionsAsString;
+    private static final Pattern SPACE_PATTERN = Pattern.compile(" ");
+
+    /**
+     * Instantiates {@code CosmosExcludedRegions}.
+     *
+     * @param excludedRegions the set of regions to exclude.
+     * @throws IllegalArgumentException if {@code excludedRegions} is set as null.
+     * */
+    public CosmosExcludedRegions(Set<String> excludedRegions) {
+
+        checkArgument(excludedRegions != null, "excludedRegions cannot be set to null");
+
+        this.excludedRegions = new HashSet<>(excludedRegions);
+        this.excludedRegionsAsString = stringifyExcludedRegions(this.excludedRegions);
+    }
+
+    /**
+     * Gets the set of excluded regions.
+     *
+     * @return a set of excluded regions.
+     * */
+    public Set<String> getExcludedRegions() {
+
+        if (this.excludedRegions == null || this.excludedRegions.isEmpty()) {
+            return new HashSet<>();
+        }
+
+        return new HashSet<>(this.excludedRegions);
```

Comment:
why a new set here? We already create a new set when initializing this?

Reply:
If we are providing a getter, I thought might as well create a copy to prevent side-effects. Also, the CX may want to perform some `Set`-based operations such as `add()` or `remove()` and set it back.